src/caf/toolkit/pandas_utils/matrices.py

```diff
@@ -0,0 +1,284 @@
+                matrix,
+                translation_factors,
+                translation_from_col,
+                translation_to_col,
+                translation_factors_col,
+            )
+
+        elif (
+            (translation_factors_col is not None)
+            or (translation_from_col is not None)
+            or (translation_to_col is not None)
+        ):
+            raise ValueError(
+                "If translation_from_col,"
+                " translation_to_col or translation_factors_col are provided,"
+                " translation must also be given"
+            )
+        else:
+            translated_describe_label = "Matrix"
+        self._matrix = matrix
+        self._translated_matrix: pd.DataFrame | None = translated_matrix
+        self._describe[translated_describe_label] = matrix_describe(matrix)
+        self._distribution: cost_utils.CostDistribution | None = None
+
+    def trip_length_distribution(
+        self,
+        cost_matrix: pd.DataFrame,
+        bins: list[int],
+    ) -> None:
+        """Calculate a distribution from the matrix passed on initilisation.
+
+        Distribution is stored within the object which can be accessed using
+        the `MatrixReport.distribution` property.
+
+        Parameters
+        ----------
+        cost_matrix : pd.DataFrame
+            Cost matrix corresponding with the inputted matrix.
+        bins : list[int]
+            Bins to use for the distribution.
+        """
+        # I have ignored Mypy complaining about index/column types as this works as intended
+        try:
+            cost_matrix.index = [int(ind) for ind in cost_matrix.index]  # type: ignore[assignment]
+            cost_matrix.columns = [int(col) for col in cost_matrix.columns]  # type: ignore[assignment]
+        except ValueError:
+            pass
+        cost_matrix = cost_matrix.loc[self._matrix.index, self._matrix.columns]  # type: ignore[index]
+        self._distribution = cost_utils.CostDistribution.from_data(
+            self._matrix.to_numpy(), cost_matrix, bin_edges=bins
+        )
+
+    def write_to_excel(
+        self,
+        writer: pd.ExcelWriter,
+        label: Optional[str] = None,
+        output_sector_matrix: bool = False,
+    ) -> None:
+        """Write the report to an Excel file.
+
+        Parameters
+        ----------
+        writer : pd.ExcelWriter
+            Excel writer to write the report with.
+        label : Optional[str], optional
+            Added to the sheet names to define the matrix, by default None.
+        output_matrix : bool, optional
+            Whether to output a sectorised matrix sheet to the Excel file, by default False.
+
+        Raises
+        ------
+        ValueError
+            If the `label` is over 30 characters long.
+        """
+
+        if label is not None:
+            sheet_prefix: str = f"{label}_"
+        else:
+            sheet_prefix = ""
+
+        if len(sheet_prefix) >= 31:
```

Comment:
This check should account for the length of the longest suffix e.g. "Distribution"
```suggestion
        if len(sheet_prefix) >= 19:
```